software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java

```diff
@@ -344,9 +346,23 @@ public void runPostLaunchCommand() {
      * @see SoftwareProcess#SHELL_ENVIRONMENT
      */
     public Map<String, String> getShellEnvironment() {
-        return Strings.toStringMap(entity.getConfig(SoftwareProcess.SHELL_ENVIRONMENT), "");
+        Map<String, Object> env = entity.getConfig(SoftwareProcess.SHELL_ENVIRONMENT);
+        if (env == null) {
+            return null;
+        }
+        ShellEnvironmentSerializer envSerializer = new ShellEnvironmentSerializer(((EntityInternal)entity).getManagementContext());
+        Map<String, String> serializedEnv = Maps.newHashMap();
+        for (Entry<String, Object> entry : env.entrySet()) {
+            String key = serializeShellEnv(envSerializer, entry.getKey());
```

Comment:
Is it really necessary to serialise the key?  It might even be worth checking here precisely that the key matches the requirements for names of environment variables?